pkg/network/state.go

```diff
@@ -62,6 +62,7 @@ type State interface {
 type Delta struct {
 	Connections []ConnectionStats
 	HTTP        map[http.Key]http.RequestStats
+	Buffer      *ClientBuffer
```

Comment:
Ideally this should be a private field, but the tracer package needs access to this to create a `Connections` object. Alternatively we can add a public method to this package like `NewConnectionsFromDelta`

Reply:
I wonder we can alias `[]ConnectionStats` to a new type, and hide the `ClientBuffer` within that type? That would keep the buffer->connections linkage close, and might prevent some of this abstraction leakage.

Reply:
I think lean towards what @ISauve mentioned, but could you elaborate on your idea, @brycekahle ? I'm not sure if I follow it

Reply:
I guess you cannot blindly embed a slice into a type. But you could do something like this:
```
type ConnectionStatsGroup struct {
	Conns []ConnectionStats
	b *ClientBuffer
}
```
and use that type instead of `[]ConnectionStats`. That way the buffer follows the slice that it corresponds to transparently, and you only need functions on `ConnectionStatsGroup` (name could probably be improved).

Reply:
I added a `BufferedData` type, though I'm not 100% positive it is worth the added complexity. Let me know what you think